bundles/org.openhab.binding.foobot/src/test/java/org/openhab/binding/foobot/internal/handler/FoobotDeviceHandlerTest.java

```diff
@@ -53,6 +54,7 @@ public void testSensorDataToState() throws IOException, FoobotApiException {
         final FoobotJsonData sensorData = connector.getSensorData("1234");
 
         assertNotNull(sensorData, "No sensor data read");
+        Objects.requireNonNull(sensorData);
```

Comment:
I'd prefer to have the null checker disabled on testing classes than be required to add unnecessary checks like this in the test code.

@wborn Can we just disable the null checker for test classes?

Reply:
For me test code is not any different as any other code. It should be clean, spotless and NPE free.

Reply:
I wonder if we annotate `Mockito.assertNonNull` the same as `Objects.requireNonNull` if that would work as well

Reply:
I'd prefer to make the barriers to entry for testing as low as possible. The more requirements you add to test code the less test code people will write. Test code should not be held to the same standard as the regular code base other than readability and maintainability. I could care less about how robust testing code is.

Also the point of using a null checker is to prevent untested runtime errors. Any test code that runs successfully can be assumed to be NPE free (unless it is really poorly written).

Just my 2 cents.

Reply:
> Also the point of using a null checker is to prevent untested runtime errors. Any test code that runs successfully can be assumed to be NPE free (unless it is really poorly written).

Based on timing due to server loads test code can also return null e.g. `getService` in itests. So tests can also behave differently at runtime. The null analysis triggers you to think about and investigate these possible null scenarios which makes tests more robust. As a result builds will less likely fail and contributors/maintainers have to spent less time debugging them and retriggering failing builds making them more happy. :-)

Reply:
Fair enough.